ui/src/components/admin/forms/AdminFormDataViewer.js

```diff
@@ -0,0 +1,252 @@
+          },
+        });
+        let formDataTemp = [];
+        data.forEach(function (item) {
+          let temp = {};
+          temp = item.submissionData;
+
+          //format date to: yyy-mm-dd hh:mm:ss
+          let dateTimeRaw = DateTime.fromISO(item.updatedAt);
+          temp.updatedAt = dateTimeRaw
+            .toISO()
+            .replace(/T/, ' ')
+            .replace(/\..+/, '');
+
+          if ('address' in temp) {
+            let addressString = [];
+            for (const property in temp['address']) {
+              addressString.push(temp['address'][property]);
+            }
+            temp['address'] = addressString.join(', ');
+          }
+          formDataTemp.push(temp);
+        });
+
+        setFormData(formDataTemp);
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    getFormData();
+  }, [formId]);
+
+  useEffect(() => {
+    if (api) {
+      if (startDate !== '' && endDate !== '' && startDate > endDate) {
+        alert('Start Date should be before End Date');
+        setEndDate('');
+      } else {
+        let dateFilterComponent = api.getFilterInstance('updatedAt');
+        dateFilterComponent.setModel({
+          type: getFilterType(),
+          dateFrom: startDate ? startDate : endDate,
+          dateTo: endDate,
+        });
+        api.onFilterChanged();
+      }
+    }
+  }, [startDate, endDate]);
+
+  useEffect(() => {
+    if (api) {
+      if (formData && formData.length > 0) {
+        api.setRowData(formData);
+        return;
+      }
+      api.setRowData([]);
+    }
+  }, [formData, api]);
+
+  const defaultColDef = {
+    editable: true,
+    sortable: true,
+    resizable: true,
+    filter: true,
+    floatingFilter: true,
+  };
+
+  const createColumnDefs = () => {
+    let columnDefs = [];
+
+    const createStringColumn = (key) => {
+      return {
+        headerName: key,
+        field: key,
+      };
+    };
+
+    const createNumberColumn = (key) => {
+      return {
+        headerName: key,
+        field: key,
+        type: 'numericColumn',
+      };
+    };
+
+    const createBooleanColumn = (key) => {
+      return {
+        headerName: key,
+        field: key,
+        cellEditor: 'agSelectCellEditor',
+        cellEditorParams: {
+          values: [true, false],
+        },
+      };
+    };
+    const createDateColumn = (key) => {
+      return {
+        headerName: key,
+        field: key,
+        filter: 'agDateColumnFilter',
+        filterParams: dateFilterParams,
+      };
+    };
+    const objectClassifier = (key, value) => {
+      if (key === 'updatedAt') {
+        columnDefs.push(createDateColumn(key));
+      } else if (typeof value === 'string') {
+        columnDefs.push(createStringColumn(key));
+      } else if (typeof value === 'number') {
+        columnDefs.push(createNumberColumn(key));
+      } else if (typeof value === 'boolean') {
+        columnDefs.push(createBooleanColumn(key, value));
```

Comment:
Your createBooleanColumn only accepts 1 argument. But you are passing 2 arguments?